app/src/main/java/com/catelt/mome/ui/components/CustomPlayerUiController.kt

```diff
@@ -0,0 +1,276 @@
+
+            override fun onStopTrackingTouch(p0: SeekBar?) {
+                youTubePlayer.play()
+                setAutoHideController()
+            }
+
+        })
+        panel.setOnClickListener {
+            if (!btnReply.isVisible){
+                setVisionControl(true)
+                setAutoHideController()
+            }
+        }
+
+        FadeViewHelper
+        setUIButtonAudio()
+    }
+
+    override fun onReady(youTubePlayer: YouTubePlayer) {
+        layoutLoading.visibility = View.GONE
+    }
+
+    private fun handlePlayPause(){
+        val isPlaying = playerTracker.state == PlayerState.PLAYING
+        if (isPlaying) {
+            youTubePlayer.pause()
+        } else {
+            youTubePlayer.play()
+        }
+    }
+
+    override fun onStateChange(youTubePlayer: YouTubePlayer, state: PlayerState) {
+        setUIButtonPlayPause(state == PlayerState.PLAYING)
+        when(state) {
+            PlayerState.PLAYING -> {
+                layoutLoading.visibility = View.GONE
+                imgBackdrop.visibility = View.GONE
+                panel.setBackgroundColor(
+                    ContextCompat.getColor(context, R.color.transparent)
+                )
+            }
+            PlayerState.PAUSED -> {
+                panel.setBackgroundColor(
+                    ContextCompat.getColor(context, R.color.transparent)
+                )
+            }
+            PlayerState.ENDED -> {
+                panel.setBackgroundColor(
+                    ContextCompat.getColor(context, R.color.black)
+                )
+                setUIReply(true)
+                imgBackdrop.visibility = View.VISIBLE
+            }
+            PlayerState.BUFFERING -> {
+                panel.setBackgroundColor(
+                    ContextCompat.getColor(context, R.color.black)
+                )
+            }
+            else -> {
+            }
+        }
+    }
+
+    private fun setUIButtonAudio() {
+        if (isMute){
+            btnAudio.setImageResource(R.drawable.ic_mute)
+            youTubePlayer.mute()
+        }else{
+            btnAudio.setImageResource(R.drawable.ic_unmute)
+            youTubePlayer.unMute()
+        }
+    }
+
+    private fun setUIReply(isReply: Boolean){
+        if (isReply){
+            btnReply.visibility = View.VISIBLE
+            btnAudio.visibility = View.GONE
+            layoutPlayPause.visibility = View.GONE
+            layoutSeekBar.visibility = View.GONE
+            seekBarProgressMini.visibility = View.GONE
+        }
+        else{
+            btnReply.visibility = View.GONE
+            btnAudio.visibility = View.VISIBLE
+            setVisionControl(false)
+        }
+    }
+
+    private fun setAutoHideController(){
+        handler.removeCallbacks(fadeOut)
+        handler.postDelayed(fadeOut, DEFAULT_TIME_DELAY)
+    }
+
+    private fun setVisionControl(isVision: Boolean){
+        if (isVision){
+            layoutPlayPause.visibility = View.VISIBLE
```

Comment:
if can, using `layoutPlayPause.isVisible = isVision` for clean code. Replace if else